### PR TITLE
Fixes #13937 - search by domain parameters searches on primary_interface

### PR DESCRIPTION
**Pull request by orrabin**

### CODE REVIEW

**Comment by dLobatog** [comment]
@orrabin Looks like sqlite3 doesn't like this syntax, otherwise :+1: , thanks !

File: app/models/concerns/hostext/search.rb [hunk 1]
```diff
@@ -162,7 +162,10 @@ def search_by_params(key, operator, value)
 
         conditions += " AND " unless conditions.blank? || negate.blank?
         conditions += " NOT(#{negate})" unless negate.blank?
-        {:conditions => conditions}
+        {
+          :joins =>  "INNER JOIN nics ON hosts.id = nics.host_id AND nics.\"primary\" = 't'",
```
Comment (orrabin) [inline app/models/concerns/hostext/search.rb]: @dLobatog locally this was the only syntax sqlite liked. If you have another suggestion I'll change this.
Edit: lzap suggested `primary` looks like a keyword (like primary key) so it has to be quoted.

Comment (tbrisker) [inline app/models/concerns/hostext/search.rb]: How about `:joins => :primary_interface`?

Comment (orrabin) [inline app/models/concerns/hostext/search.rb]: That works, changing this.

**Comment by iNecas** [comment]
[test]

**Comment by dLobatog** [comment]
Merged as f2506d3482098dcd4faffe64c036f139d77034d7, thanks @orrabin!

**Pull request closed without merging**